app/frontend/src/components/Navigation/Navigation.tsx

```diff
@@ -119,38 +156,67 @@ export default function Navigation() {
           gutters: classes.gutters,
         }}
       >
-        <Hidden smDown>
+        <div className={classes.nav}>
+          <Hidden mdUp>
+            <IconButton
+              className={classes.icon}
+              aria-label="open drawer"
+              onClick={handleDrawerOpen}
+              edge="start"
+            >
+              <MenuIcon />
+            </IconButton>
+            <Drawer
+              variant="temporary"
+              anchor="left"
+              open={open}
+              ModalProps={{
+                keepMounted: true, // better open performance on mobile
+              }}
+              classes={{
+                paper: classes.drawerPaper,
+              }}
+            >
+              <div className={classes.drawerHeader}>
+                <Typography variant="h5" className={classes.drawerTitle}>
+                  {COUCHERS}
+                </Typography>
+                <IconButton
+                  className={classes.icon}
+                  aria-label="close drawer"
+                  onClick={handleDrawerClose}
+                >
+                  <CloseIcon />
+                </IconButton>
+              </div>
+              {drawer}
+            </Drawer>
+          </Hidden>
           <Typography variant="h5" className={classes.title}>
-            Couchers
+            {COUCHERS}
           </Typography>
-        </Hidden>
-        <Grid
-          container
-          wrap="nowrap"
-          classes={{
-            root: classes.flex,
-          }}
-        >
-          {menu.map((item) => (
-            <NavButton
-              route={item.route}
-              label={item.name}
-              key={`${item.name}-nav-button`}
+          <Hidden mdDown>
+            <Grid
```

Comment:
This `<Grid container>` has no `<Grid item>` children, so maybe it can just be a div?